### PR TITLE
[CONTINT-3554] Fix build on linux and do not send cgroup node inode if in host cgroup namespace

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR adds build flags in the logic that retrieves the container id since it only works on linux.
It also only retrieves the cgroup node inode if the sender not in the host cgroup namespace. Indeed it would mean either that:
- We're not in a container and the inode we're sending is not related to a container
- We already have access to the container-id and it will be retrieved in the previous logic

### Motivation
Fix build and do not emit data if not necessary (limit perf impact)

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: internal/container_linux_test.go [hunk 1]
```diff
@@ -3,6 +3,8 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016 Datadog, Inc.
 
+//go:build test && linux
```
Comment (nsrip-dd) [inline internal/container_linux_test.go:6]: The `test` build tag isn't defined, so these tests won't run. I think you just want the `linux` tag.

**Review by nsrip-dd — CHANGES_REQUESTED** [review]
Thanks! One change needed for the test file build tags (see my comment). Should someone with cgroups knowledge review the other changes?

File: internal/container_linux_test.go (hunk 1 above)
Comment (AliDatadog) [reply internal/container_linux_test.go]: Thanks. I just removed it !

I also asked a review to my team members. They should have the knowledge necessary. Normally I am just aligning on what is done in dogstatsd https://github.com/DataDog/datadog-go/pull/291

**Review by nsrip-dd — APPROVED** [review]

**Comment by nsrip-dd** [comment]
Changes look similar to the linked datadog-go changes, which were approved by someone with more knowledge, so LGTM

**Pull request merged**